tests/std/include/range_algorithm_support.hpp

```diff
@@ -936,6 +949,11 @@ constexpr void test_contiguous() {
     with_contiguous_ranges<Instantiator, Element>::call();
 }
 
+template <class Instantiator, class Element1, class Element2>
+constexpr void test_in_outerator() {

```

Comment:
I apologize for this truly terrible name. Suggestions? We can't simply use `test_in_out` because `out` already means "models of `output_range`" for these functions.

Reply:
Perhaps the test code could refer to such iterators as hybrids or chimeras?

Reply:
This function template calls `Instantiator::template call<X, Y>()` with many types `X` that model `input_range` (having elements of type `Element1`) and many types `Y` that model `output_iterator<Z>` iff `declval<Element2&>() = declval<Z>()` is well-formed. There are actually few Ranges algorithms that require models of `output_iterator`, most actually require what I refer to as "writable iterators" which are output iterators minus the `*i++ = meow` requirement. We have another template `test_in_writable` that generates `input_range`/"writable iterator" combinations, and at least `test_out` that generates `output_range`s. 

So the iterators themselves that are special here. The problem is that we need something to put in the name of this function to suggest that the second type in each combination it generates is an output iterator type, not an output range type. "outerator" does that poorly. The best fix is probably to include the iterator vs. range distinction in the names of all of these functions so this would be something like `input_range_output_iterator_permutations`.